### ISSUE TITLE
PHP[WARNING]: Array to string conversion /usr/local/www/snappymail/snappymail/v/2.36.1/app/libraries/snappymail/gpg/pgp.php [line:879, code:2]

### ISSUE THREAD

**Issue by yasuhirokimura** [the-djmaze/snappymail#1560]
Under following conditions,

* FreeBSD 14.0-RELEASE amd64
* Dovecot 2.3.21
* Apache HTTP server 2.4.59
* PHP 8.3.6
* SnappyMail 2.36.1

Following lines are written to log file.

```
[2024-04-25 01:26:14.028][76648a1f] PHP[WARNING]: Array to string conversion /usr/local/www/snappymail/snappymail/v/2.36.1/app/libraries/snappymail/gpg/pgp.php [line:879, code:2]
[2024-04-25 01:26:14.030][76648a1f] PHP[WARNING]: Array to string conversion /usr/local/www/snappymail/snappymail/v/2.36.1/app/libraries/snappymail/gpg/pgp.php [line:879, code:2]
[2024-04-25 01:26:14.032][76648a1f] OB-DATA[ERROR]: <br />
<b>Warning</b>:  Array to string conversion in <b>/usr/local/www/snappymail/snappymail/v/2.36.1/app/libraries/snappymail/gpg/pgp.php</b> on line <b>879</b><br />
<br />
<b>Warning</b>:  Array to string conversion in <b>/usr/local/www/snappymail/snappymail/v/2.36.1/app/libraries/snappymail/gpg/pgp.php</b> on line <b>879</b><br />
```

**Comment by the-djmaze** [comment]
Sounds like a bug in PHP 8.3 as older versions don't throw this error.
Secondly the code of SnappyMail c9nforms the documentation at https://php.net/proc_open

Will see if i can duplicate the issue

**Comment by daffydock** [comment]
Yeah, I am getting something similar on NC logs. Just after I updated to NC 29. It happens whenever SnappyMail starts up. App does start, however. Currently on PHP 8.1, though. 

Error:

`explode():` Passing null to parameter #2 ($string) of type string is deprecated at /home/[USER]/[DOMAIN]/apps/snappymail/app/snappymail/v/2.36.1/app/libraries/snappymail/gpg/pgp.php#464 

Posting here, as it may be related.

**Comment by Marcus1Pierce** [comment]
@the-djmaze I have this error too on Nextcloud Hub 6 (27) with PHP Version 8.2.18

My error on Nextcloud:
```
[PHP] Error: Array to string conversion at /var/www/html/custom_apps/snappymail/app/snappymail/v/2.36.1/app/libraries/snappymail/gpg/pgp.php#879

POST /apps/snappymail/?/Json/&q[]=/0/
from 36.65.245.1 by purchasing at 2024-04-26T01:15:00+00:00
```

**Comment by the-djmaze** [comment]
There is definitely something wrong with PHP.

```
proc_open(
    array|string $command,
    array $descriptor_spec,
    array &$pipes,
    ?string $cwd = null,
    ?array $env_vars = null,
    ?array $options = null
): resource|false
```

As you can see only 2 parameters are string.
SnappyMail has:
* $command is a string
* $cwd = null

So there's no way that error can occur on that line.
https://github.com/the-djmaze/snappymail/blob/9ec8ac8bbab56af13187fef5b7cda0f3eb88f9ff/snappymail/v/0.0.0/app/libraries/snappymail/gpg/pgp.php#L879-L886

**Comment by Marcus1Pierce** [comment]
@the-djmaze How can i test the code if i install snappymail on nextcloud docker? And what should i modify, change or edit on that file?

**Comment by truthsword** [comment]
I'm seeing this quote often as well...
```
[2024-05-31 18:06:03.085][c7c38b1e] PHP[WARNING]: Array to string conversion /snappymail/snappymail/v/2.36.3/app/libraries/snappymail/gpg/pgp.php [line:873, code:2]`

NOTICE: PHP message: PHP Warning:  Array to string conversion in /snappymail/snappymail/v/2.36.3/app/libraries/snappymail/gpg/pgp.php on line 873
```
In addition, I have this frequently repeated:

`gpgconf: Note: '--homedir' is not considered an option`

I'n unsure if the two are related.

**Comment by the-djmaze** [comment]
> In addition, I have this frequently repeated:
> 
> `gpgconf: Note: '--homedir' is not considered an option`
> 
> I'n unsure if the two are related.

no, that looks like an old gpgconf as the option is available in gpgconf 2.4

**Comment by truthsword** [comment]
> that looks like an old gpgconf as the option is available in gpgconf 2.4

I cannot locate gpgconf in the container. Where would that be?

**Comment by the-djmaze** [comment]
> I cannot locate gpgconf in the container. Where would that be?

No idea, but `which gpgconf` should give you a hint.
